moabb/datasets/dreyer2023.py

```diff
@@ -0,0 +1,547 @@
+"""
+A large EEG right-left hand motor imagery dataset.
+It is organized into three A, B, C datasets.
+URL PATH: https://zenodo.org/record/7554429
+"""
+
+import warnings
+import zipfile
+from pathlib import Path
+
+import pandas as pd
+from mne_bids import BIDSPath, get_entity_vals, read_raw_bids
+from tqdm import tqdm
+
+from moabb.datasets import download as dl
+
+from .base import BaseDataset
+
+
+_manifest_link = "https://osf.io/download/67c9abecc1b99765d8bb36b0/"
+
+_osf_tag = "8tdk5"
+_api_base_url = f"https://files.de-1.osf.io/v1/resources/{_osf_tag}/providers/osfstorage/"
+
+
+class _Dreyer2023Base(BaseDataset):
+    """
+    Parent class of Dreyer2023A, Dreyer2023B and Dreyer2023C.
+    Should not be instantiated.
+    """
+
+    def __init__(self, subjects, sub_id="A"):
+
+        self.sub_id = sub_id
+        self.sub_id_shift = dict(A=0, B=60, C=81)
+
+        if sub_id is None:
+            self.subject_list = subjects
+            self.sub_id = ""
+        else:
+            self.subject_list = [sub_id + str(i) for i in subjects]
```

Comment:
I'm not a huge fan of this... 
Everywhere in the moabb API, the subjects are integers, not str. I think we should stay consistent with this.

It's ok if dataset B contains subjects `range(60, 81)`, they don't have to start at 1.